fl4health/clients/moon_client.py

```diff
@@ -117,19 +141,38 @@ def compute_loss(
             target: (torch.Tensor): Ground truth data to evaluate predictions against.
 
         Returns:
-            Losses: Object containing checkpoint loss, backward loss and additional losses indexed by name.
+            TrainingLosses: an instance of TrainingLosses containing backward loss and additional losses
+            indexed by name.
         """
         if len(self.old_models_list) == 0:
-            return super().compute_loss(preds, features, target)
-        loss = self.criterion(preds["prediction"], target)
-        total_loss = loss.clone()
-        additional_losses = {}
+            return super().compute_training_loss(preds, features, target)
 
-        if self.contrastive_weight:
-            contrastive_loss = self.get_contrastive_loss(
-                features["features"], features["global_features"], features["old_features"]
-            )
-            total_loss += self.contrastive_weight * contrastive_loss
-            additional_losses["contrastive_loss"] = contrastive_loss
-        losses = Losses(checkpoint=loss, backward=total_loss, additional_losses=additional_losses)
-        return losses
+        total_loss, additional_losses = self.compute_loss_and_additional_losses(preds, features, target)
+        return TrainingLosses(backward=total_loss, additional_losses=additional_losses)
+
+    def compute_evaluation_loss(
+        self,
+        preds: Dict[str, torch.Tensor],
+        features: Dict[str, torch.Tensor],
+        target: torch.Tensor,
+    ) -> EvaluationLosses:
+        """
+        Computes evaluation loss given predictions and features of the model and ground truth data. Loss includes
+        base loss plus a model contrastive loss.
+
+        Args:
+            preds (Dict[str, torch.Tensor]): Prediction(s) of the model(s) indexed by name.
+                All predictions included in dictionary will be used to compute metrics.
+            features: (Dict[str, torch.Tensor]): Feature(s) of the model(s) indexed by name.
+            target: (torch.Tensor): Ground truth data to evaluate predictions against.
+
+        Returns:
+            EvaluationLosses: an instance of EvaluationLosses containing checkpoint loss and
+                additional losses indexed by name.
+        """
+        if len(self.old_models_list) == 0:
+            return super().compute_evaluation_loss(preds, features, target)
+
+        loss = self.criterion(preds["prediction"], target)
```

Comment:
Similar to the FENDA example, perhaps you can just store the loss in additional losses and extract from there. You could also pull out the total loss and put it into additional losses as well.

Reply:
Not sure if what I did is what you had in mind, please check.